js/src/forum/addFlagControl.js

```diff
@@ -7,7 +7,7 @@ import FlagPostModal from './components/FlagPostModal';
 
 export default function() {
   extend(PostControls, 'userControls', function(items, post) {
-    if (post.isHidden() || post.contentType() !== 'comment' || !post.canFlag() || post.user() === app.session.user) return;
+    if (post.isHidden() || post.contentType() !== 'comment' || !post.canFlag() || post.user() === app.session.user && !Number(app.forum.attribute('canFlagOwn'))) return;
```

Comment:
I am missing a server-side validation of this flag.

:thinking: 

Also, could we just add that into the server-generated value of each post's `canFlag` attribute? That way, we'd have less logic on the client-side.